### PR TITLE
Enable AWS in KEB

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- KEB with AWS

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/provider/aws_provider.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+func (p *AWSInput) Profile() gqlschema.KymaProfile {
+	return gqlschema.KymaProfileProduction
+}
+
+func (p *AWSTrialInput) Defaults() *gqlschema.ClusterConfigInput {
+	return &gqlschema.ClusterConfigInput{
+		GardenerConfig: &gqlschema.GardenerConfigInput{
+			DiskType:       "Standard_LRS",
```
Comment (tobiscr) [inline components/kyma-environment-broker/internal/provider/aws_provider.go:96]: Shouldn't it be `GP2`?

File: components/kyma-environment-broker/internal/provider/aws_provider.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+			VolumeSizeGb:   50,
+			MachineType:    "Standard_D4_v3",
```
Comment (tobiscr) [inline components/kyma-environment-broker/internal/provider/aws_provider.go:98]: `			MachineType:    "m4.xlarge",`

File: components/kyma-environment-broker/internal/provider/aws_provider.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	AWSInput      struct{}
+	AWSTrialInput struct {
+		PlatformRegionMapping map[string]string
+	}
+)
+
+func (p *AWSInput) Defaults() *gqlschema.ClusterConfigInput {
+	return &gqlschema.ClusterConfigInput{
+		GardenerConfig: &gqlschema.GardenerConfigInput{
+			DiskType:       "Standard_LRS",
```
Comment (tobiscr) [inline components/kyma-environment-broker/internal/provider/aws_provider.go:36]: Should be GP2

**Review by tobiscr — CHANGES_REQUESTED** [review]

File: components/kyma-environment-broker/internal/process/provisioning/create_runtime.go [hunk 4]
```diff
@@ -56,11 +56,15 @@ func (s *CreateRuntimeStep) Run(operation internal.ProvisioningOperation, log lo
...
+		log.Infof("call ProvisionRuntime: kymaVersion=%s, kubernetesVersion=%s, region=%s, kymaProfile=%s, provider=%s",
 			requestInput.KymaConfig.Version,
 			requestInput.ClusterConfig.GardenerConfig.KubernetesVersion,
 			requestInput.ClusterConfig.GardenerConfig.Region,
-			requestInput.KymaConfig.Profile)
+			requestInput.KymaConfig.Profile,
+			requestInput.ClusterConfig.GardenerConfig.Provider)
+
+		log.Debugf("Provider specific config %+v", requestInput.ClusterConfig.GardenerConfig.ProviderSpecificConfig)
+		log.Debugf("Provider specific config aws %+v", requestInput.ClusterConfig.GardenerConfig.ProviderSpecificConfig.AwsConfig)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/create_runtime.go:67]: todo: remove it

**Review by tobiscr — DISMISSED** [review]

File: components/kyma-environment-broker/cmd/broker/main.go [hunk 5]
```diff
@@ -158,6 +160,10 @@ func main() {
 
 	logs := logrus.New()
 	logs.SetFormatter(&logrus.JSONFormatter{})
+	if cfg.LogLevel != "" {
```
Comment (jasiu001) [inline components/kyma-environment-broker/cmd/broker/main.go:163]: as I understand it is done by the way ;)

File: components/kyma-environment-broker/internal/broker/plans.go [hunk 6]
```diff
@@ -80,6 +84,23 @@ func GCPSchema(machineTypes []string) []byte {
...
+	schema := NewSchema(properties, DefaultControlsOrder())
+
+	bytes, err := json.Marshal(schema)
+	if err != nil {
+		panic(err)
+	}
+	return bytes
+}
+
+func AWSRegions() []string {
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/broker/plans.go:98]: nit: all regions are at the top (azure, gcp)

File: components/kyma-environment-broker/internal/process/input/builder.go [hunk 7]
```diff
@@ -78,8 +79,9 @@ func NewInputBuilderFactory(optComponentsSvc OptionalComponentService, disabledC
 }
 
 func (f *InputBuilderFactory) IsPlanSupport(planID string) bool {
+	logrus.Infof(planID)
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/process/input/builder.go:82]: is this debugging residue?

File: components/kyma-environment-broker/internal/process/input/input.go [hunk 8]
```diff
@@ -20,12 +20,13 @@ const (
...
-	MachineImage                string        `envconfig:"optional"`
-	MachineImageVersion         string        `envconfig:"optional"`
-	TrialNodesNumber            int           `envconfig:"optional"`
+	Timeout                     time.Duration               `envconfig:"default=12h"`
+	KubernetesVersion           string                      `envconfig:"default=1.16.9"`
+	DefaultGardenerShootPurpose string                      `envconfig:"default=development"`
+	MachineImage                string                      `envconfig:"optional"`
+	MachineImageVersion         string                      `envconfig:"optional"`
+	TrialNodesNumber            int                         `envconfig:"optional"`
+	DefaultTrialProvider        internal.TrialCloudProvider `envconfig:"default=Azure"` // could be: Azure, AWS, GCP
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/process/input/input.go:29]: where it is used?

**Comment by jasiu001** [comment]
add Plan description/metadata to the [catalog.yaml](https://github.com/kyma-project/control-plane/blob/master/resources/kcp/charts/kyma-environment-broker/files/catalog.yaml#L40) file.

don't forget bump KEB image

File: components/kyma-environment-broker/cmd/broker/main.go (hunk 5 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/main.go:163]: yes, I realized it is missing

File: components/kyma-environment-broker/internal/process/input/builder.go (hunk 7 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/input/builder.go:82]: left by accident, removed

File: components/kyma-environment-broker/internal/process/input/input.go (hunk 8 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/process/input/input.go:29]:
```func (f *InputBuilderFactory) forTrialPlan(provider *internal.TrialCloudProvider) HyperscalerInputProvider {
	var trialProvider internal.TrialCloudProvider
	if provider == nil {
		trialProvider = f.config.DefaultTrialProvider
	} else {
		trialProvider = *provider
	}```

**Review by jasiu001 — APPROVED** [review]

**Pull request merged**